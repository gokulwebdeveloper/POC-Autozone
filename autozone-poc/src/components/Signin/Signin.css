/* Signin page css */

.signinhead{
    padding: 1rem;
}
.link{
    text-decoration: underline;
    color:blue ;
    text-align: left;
    margin-left: 87%;
    font-size: 1rem;
}
.signincontainer{
    border: solid 1px #d0d0dc;
    border-radius: 20px;
    margin-top: 1%;
    margin-bottom: 1%;
}
.btn{
    border-radius: 10px;
}