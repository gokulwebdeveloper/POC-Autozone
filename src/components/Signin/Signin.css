/* Signin page css */

.signinhead{
    padding: 1rem;
}
.link{
    text-decoration: underline;
    color:blue ;
    text-align: left;
    margin-left: 87%;
    font-size: 1rem;
}
.signincontainer{
    color: aliceblue;
    border: solid 1px #d0d0dc;
    border-radius: 7px;
    margin-top: 1%;
    margin-bottom: 1%;
    margin-left: 19%;
    background-color: #2986d6de;
}
.signinExplanation{
    padding: 1rem;
    color: aliceblue;
}
.btnpostion{
    text-align: center;
}
.bgimage{
    background-image: url( "../../assets/img/bg-img/loginbg.jpg");
    height: 100%;
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
}