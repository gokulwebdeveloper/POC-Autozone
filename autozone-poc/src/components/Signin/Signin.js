import React, { Fragment, useState } from "react";
import axios from "axios";

import InputWithLabel from "../../utlis/formComponents/InputWithLabel";
import "./Signin.css";
import Button from "../../utlis/button/Button";
import { data } from "jquery";

export default function Signin() {
  const [emailid, setEmail] = useState("");
  const [password, setPassword] = useState("");


  const onSubmit = (evt) => {
    const data = {};
    evt.preventDefault();
    data.emailid = emailid;
    data.password = password;
    axios.post("http://localhost:5000/user/signin", data).then((response) => {
      console.log(response);
    });
  };
  return (
    <div>
      <div className="container h-100">
        <div className="row h-100 align-items-center">
          <div className="col-4"></div>
          <div className="col-4 align-items-center signincontainer">
            <div className="page-title text-center">
              <h2 className="signinhead">Sign In</h2>

              <form onSubmit={onSubmit}>
              
                <InputWithLabel
                  labelText="Email Address"
                  inputId="email_address"
                  inputType="email"
                  onChange={(e) => setEmail(e.target.value)}
                  isRequired={true}
                />
                <InputWithLabel
                  labelText="Password"
                  inputId="password"
                  type="password"
                  onChange={(e) => setPassword(e.target.value)}
                  isRequired={true}
                />
                <br />
                <br />
                <div className="link">
                  <a href="/signup">Signup</a>
                </div>
                <br />
                <Button btnText="Signin" type="submit" />
                <br />
                <br />
              </form>
            </div>
          </div>
          <div className="col-4"></div>
        </div>
      </div>
    </div>
  );
}
