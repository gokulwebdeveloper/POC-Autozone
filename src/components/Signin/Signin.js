import React, { Fragment, useState } from "react";
import axios from "axios";
import { BrowserRouter as Router, NavLink, Link } from "react-router-dom";
import InputWithLabel from "../../utlis/formComponents/InputWithLabel";
import "./Signin.css";
import Button from "../../utlis/button/Button";
//import { data } from "jquery";

export default function Signin() {
  const [emailid, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [msgerror, setmsgerror] = useState(false);

  const onSubmit = (evt) => {
    const data = {};
    evt.preventDefault();
    data.emailid = emailid;
    data.password = password;
    axios
      .post("http://localhost:5000/user/signin", data)
      .then((response) => {
        localStorage.setItem("accessToken", response.data.accessToken);
        window.location.assign("/");
      })
      .catch((err) => {
        data.msgerror = true;
        window.scrollTo(0, 0);
        setmsgerror(data.msgerror);
      });
  };
  return (
    <div>
      <div className="container-fluid h-100 bgimage">
      {msgerror ? (
        <div className="row justify-content-md-center">
          <div className="alert alert-danger col-md-4" role="alert">
            invalid Email id or password!
          </div>
        </div>
      ) : null}
        <div className="row h-100 align-items-center">
          <div className="col-md-5">
            <h1 className="signinExplanation">
              WELCOME BACK Sign In to save your vehicles, track your orders, and
              earn Rewards when you purchase.
            </h1>
          </div>
          <div className="col-md-4  signincontainer">
            <div className="page-title">
              <h2 className="signinhead">Sign In</h2>

              <form onSubmit={onSubmit}>
                <InputWithLabel
                  labelText="Email Address"
                  inputId="email_address"
                  inputType="email"
                  onChange={(e) => setEmail(e.target.value)}
                  isRequired={true}
                />
                <InputWithLabel
                  labelText="Password"
                  inputId="password"
                  inputType="password"
                  onChange={(e) => setPassword(e.target.value)}
                  isRequired={true}
                />
                <br />
                <br />
                <div className="link">
                <NavLink
                to="/signup"
                to={{
                  pathname: "/signup",
               
                }}
              >Signup

              </NavLink>
                  
                </div>
                <br />
                <div className="btnpostion">
                  <Button btnText="Signin" type="submit" />
                </div>
                <br />
                <br />
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
