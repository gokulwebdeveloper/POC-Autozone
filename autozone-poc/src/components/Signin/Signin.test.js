import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Signin from "./Signin";

let page;
//mount component for testing
beforeEach(() => {
  page = mount( <Signin />);
});

//Cleanup after test execution
afterEach(() => {
  page.unmount();
});

describe("test Signin Page", () => {
  //test if Signin component exists
  it("Signin page Exists", () => {
    expect(page.exists()).toBeTruthy();
  });

  //if it has title as "Sign in"
  it('has titles as "Sign In"', () => {
    expect(page.find("h2").first().text()).toBe("Sign In");
  });


  //if it has 1 Button componenent
  it("has 1 Button componenent", () => {
    expect(page.find("Button")).toHaveLength(1);
  });

  //if Button has text Signin
  it("Button has text Signin", () => {
    expect(page.find("Button").text()).toBe("Signin");
  });

  //if it has 1 form for Signin
  it("has 1 form for Signin", () => {
    expect(page.find("form")).toHaveLength(1);
  });

  //SnapShot Match test
  it("SnapShot Match", () => {
    const page = renderer.create(<Signin />).toJSON();
    expect(page).toMatchSnapshot();
  });
});
