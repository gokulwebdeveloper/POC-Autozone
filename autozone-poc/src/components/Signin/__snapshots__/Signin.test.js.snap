// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Signin Page SnapShot Match 1`] = `
<div>
  <div
    className="container h-100"
  >
    <div
      className="row h-100 align-items-center"
    >
      <div
        className="col-4"
      />
      <div
        className="col-4 align-items-center signincontainer"
      >
        <div
          className="page-title text-center"
        >
          <h2
            className="signinhead"
          >
            Sign In
          </h2>
          <form
            onSubmit={[Function]}
          >
            <label
              htmlFor="email_address"
            >
              Email Address
               
              <span>
                *
              </span>
            </label>
            <input
              className="form-control"
              defaultValue=""
              id="email_address"
              onBlur={[Function]}
              onChange={[Function]}
              required={true}
              type="email"
            />
            <p
              className="formErrorMsg"
            />
            <label
              htmlFor="password"
            >
              Password
               
              <span>
                *
              </span>
            </label>
            <input
              className="form-control"
              defaultValue=""
              id="password"
              onBlur={[Function]}
              onChange={[Function]}
              required={true}
              type="password"
            />
            <p
              className="formErrorMsg"
            />
            <br />
            <br />
            <div
              className="link"
            >
              <a
                href="/signup"
              >
                Signup
              </a>
            </div>
            <br />
            <button
              className="btn essence-btn"
              type="submit"
            >
              Signin
            </button>
            <br />
            <br />
          </form>
        </div>
      </div>
      <div
        className="col-4"
      />
    </div>
  </div>
</div>
`;
